src/main/java/com/twilio/type/SubscribeRule.java

```diff
@@ -0,0 +1,271 @@
+        }
+
+        @JsonCreator
+        public static Type forValue(final String value) {
+            return Promoter.enumFromString(value, Type.values());
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                    .add("type", this.type)
+                    .toString();
+        }
+    }
+
+    public enum Kind {
+        AUDIO("audio"),
+        DATA("data"),
+        VIDEO("video");
+
+        private final String kind;
+
+        Kind(final String kind) {
+            this.kind = kind;
+        }
+
+        @JsonCreator
+        public static Kind forValue(final String value) {
+            return Promoter.enumFromString(value, Kind.values());
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                    .add("kind", this.kind)
+                    .toString();
+        }
+    }
+
+    public enum Priority {
+        LOW("low"),
+        MEDIUM("medium"),
+        HIGH("high");
+
+        private final String priority;
+        Priority(final String priority) {
+            this.priority = priority;
+        }
+
+        @JsonCreator
+        public static Priority forValue(final String value) {
+            return Promoter.enumFromString(value, Priority.values());
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                    .add("priority", this.priority)
+                    .toString();
+        }
+    }
+
+    private static final SubscribeRule subscribeAll = builder().withAll(true).withType(Type.INCLUDE).build();
+    private static final SubscribeRule subscribeNone = builder().withAll(true).withType(Type.EXCLUDE).build();
+
+    @JsonProperty("type")
+    private final Type type;
+
+    @JsonProperty("all")
+    private final Boolean all;
+
+    @JsonProperty("publisher")
+    private final String publisher;
+
+    @JsonProperty("track")
+    private final String track;
+
+    @JsonProperty("kind")
+    private final Kind kind;
+
+    @JsonProperty("priority")
+    private final Priority priority;
+
+    public SubscribeRule(@JsonProperty("type") final Type type,
+            @JsonProperty("all") final Boolean all,
+            @JsonProperty("publisher") final String publisher,
+            @JsonProperty("track") final String track,
+            @JsonProperty("kind") final Kind kind,
+            @JsonProperty("priority") final Priority priority) {
+        this.type = type;
+        this.all = all;
+        this.kind = kind;
+        this.priority = priority;
+        this.publisher = publisher;
+        this.track = track;
+    }
+
+    public SubscribeRule() {
+        this.type = null;
+        this.all = null;
+        this.kind = null;
+        this.priority = null;
+        this.publisher = null;
+        this.track = null;
+    }
+
+    @JsonIgnore
+    public boolean isValid() {
```

Comment:
Logic looks correct to me, but really needs tests to help with future maintenance.